src/components/CommerceListItem.js

```diff
@@ -1,12 +1,25 @@
 import React, { Component } from 'react';
-import { View } from 'react-native';
 import { ListItem, Button } from 'react-native-elements';
 import Icon from 'react-native-vector-icons/MaterialIcons';
-
+import {
+  registerFavoriteCommerce,
+  deleteFavoritesCommerces,
+  readFavoriteCommerce
+} from '../actions/CommercesListActions';
+import { connect } from 'react-redux';
 class CommerceListItem extends Component {
   state = { favorite: false };
 
-  onFavoritePress = () => {
+  componentWillMount() {
+    this.setState({ favorite: this.props.favoritesCommerce.includes(this.props.commerce.id) });
+  }
+
+  onFavoritePress = commerceId => {
+    if (this.props.favoritesCommerce.includes(commerceId)) {
```

Comment:
aca en vez de validar eso en el if, podes usar el valor this.state.favorite que definis en el componentWillMount